Materialize camera photo point forms

alxngyn opened the pull request:
<!--
  This is a guideline for what a PR should look like.
  Feel free to modify it to fit your specific needs.

  Please list the issue this fixes in the PR
-->
fixes issue #200 

## Changes in this PR.
<!--
  Please include a list of all things this PR will accomplish
  Use the checkbox syntax to show what is done and what is yet to be completed.
  This gives context to the diff.
-->

- [x] Materialize camera point datasheet add page `camera/edit`
- [x] materialize camera point datasheet view `camera/<camera-id>`
- [x] Materialize photo point view `camera/<camera_id>/photo/<photo_id>`
- [x] Materialize photo point edit view `camera/<camera_id>/photo/edit`

## Testing this PR.
<!--
  Please include a list of explicit instructions for testing this PR.
  If the instructions are 'run `make test`' say that,
  If they are more complicated be thorough.
-->

1. Do `docker-compose up`.
2. Do look at a camera photo point add page and make one
3. Do look at a camera photo point view page of the abomination you made
4. Do look at the photo point detail page of from the camera point view page
5. Do Try to edit a photo point page 

9000. test em all.

### Expected Output.
<!--
  Please insert either a description of what should happen when testing
  your PR or a code-block with terminal output.
-->

SSs coming soon ... 

@osuosl/devs

alxngyn commented:
@leian7 is the location field necessary since the form is based on a site?

leian7 commented:
@alxngyn Yes since one site can have multiple camera points... not sure whether the field should be user-entered coordinates or map.

leian7 commented:
Looks good to me!!

I think #200 might also include the edit-view and the view-view for individual photo points though? Like http://localhost:8000/sites/shafer-creek/camera/1/photo/edit and http://localhost:8000/sites/shafer-creek/camera/1/photo/1 as example urls. For the edit you can upload one new photo for the given photo point. For the view you can view all of that photo point's photos. Maybe this can be addressed in a separate PR anyway.

LyonesGamer commented:
Mostly looks good, although I'd go ahead and set a maximum size on the images themselves on the pp view page, because right now it shows it as whatever the image's original resolution is, which can make it appear huge on the page.

alxngyn commented:
@LyonesGamer 
Will do on the image resizing.

@leian7 
Seems like the edit form and the view form for photopoints are not connected. I believe this is technically part of the PR since i'm doing the whole data sheet. I'll do that

alxngyn commented:
@leian7 
Could you tell me the purpose of `photo_point_add.html`? I'm trying to figure out where this ties in with the user or is it a form for admins? I believe I should put a link to it somewhere but not sure if its more related to camera point activities or photo point activities.
ex routing: 
http://localhost:8000/sites/shafer-creek/camera/2/photo/edit

leian7 commented:
Yes, a good question... basically a site can have multiple camera points
and a camera point can have multiple photo points beyond the first three
that the user provides when they go to create a new camera point. In other
words, for a given camera point, you can add a fourth, fifth, etc. photo
point one by one by using photo point add view.

It should be similar to how the old site set up camera point/photo point
adding. I will forward you an email from one of the StreamWebs people that
talks about it.

On Wed, Mar 15, 2017 at 3:25 PM, Alex Nguyen <notifications@github.com>
wrote:

> @leian7 <https://github.com/leian7>
> Could you tell me the purpose of photo_point_add.html? I'm trying to
> figure out where this ties in with the user or is it a form for admins? I
> believe I should put a link to it somewhere but not sure if its more
> related to camera point activities or photo point activities.
> ex routing:
> http://localhost:8000/sites/shafer-creek/camera/2/photo/edit
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/osuosl/streamwebs/pull/219#issuecomment-286899203>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIu_SRyw_jBG5ohvFG0C5wA_XbviT2uEks5rmGVsgaJpZM4L7hYj>
> .
>

leian7 commented:
Can you add the date widget for the form on camera/<camera_id>/photo/edit and the mini-form on camera/<camera_id>/photo/<photo_id> too?

alxngyn commented:
@leian7 fixed

leian7 commented:
Thanks-- not urgent but can you also container-ize ``http://localhost:8000/sites/[site]/camera/[id]/photo/edit`` and ``http://localhost:8000/sites/[site]/camera/[id]/photo/[id]``? Otherwise +1

alxngyn commented:
@leian7 
Containerized

@athai 
> Day and month are being switched during cp date input (e.g. 2017-04-17 is being sent as 2017-17-
> 04 causing the form to error)

Fixed
> Photo point dates are being stored as the current day's date, regardless of what you're setting it as

After investigating, it's because there is no date input for photopoints on the form and it's defaulting to "today". The date input at the top is for the camera point, the date inputs at the bottom is for each of the PP image. Should we add one more date input for the point or should it take some value?

> After submitting a form successfully, you're unable to view the site detail page. This seems to be because of the above error.

Fixed
> 
> Additionally,
> 
> Submit button color needs to be updated to match the rest of the theme (I believe we're using teal-darken-4)

Fixed

> Please change Cp Date to read Camera Point Date

Fixed 

> Also, it's not really an issue, but is there any reason why you're using Google Earth rather than Google maps?

I'm using Google maps API. Where does it say Google Earth?

athai commented:
@alxngyn My wording was unclear. What I meant was, is there a reason why the map is displayed as
![ge_location](https://cloud.githubusercontent.com/assets/6801364/25150750/da89857c-2437-11e7-902d-8340d5e1ab1e.png)
for the photo point monitoring data sheet rather than how it appears on the site detail page?
![gmaps_default](https://cloud.githubusercontent.com/assets/6801364/25150991/baaede5e-2438-11e7-936c-928eba9343bb.png)

It's not a problem, but I'm curious if you had intended for there to be a difference? The submit button's color hasn't been fixed by the way.

athai commented:
@LyonesGamer @alxngyn It'd make sense if the photo point date fields defaulted to match the camera point. I don't think there's a case in which we'd want the photo point dates to differ from its parent object.

leian7 commented:
@LyonesGamer @alxngyn @athai Is this for the camera-point-add form? Then yeah, the dates for the three photo points that are immediately added for that camera point would most likely always share the same date with the camera point's date. But ultimately, it should be possible for photo point dates to be later than camera point dates-- you could record data for the new camera point and two new photo points on the same day, then come back the next day to collect the data for the third. This scenario is more likely for the individual photo-point-adds, though.

Another issue this brings up is whether we should be tracking/organizing information by both "date-the-data-was collected-in-the-field" and "date-the-data-was-entered-and-submitted-online."

For this particular data sheet, google Earth instead of google Maps might actually be ok since camera point coordinates are going to be very close to Site coordinates. It may be helpful to students if they can visually identify the exact patches of ground within their Site that they set up their camera point at... but that's only assuming they're able to zoom in that far.

alxngyn commented:
@athai @LyonesGamer @leian7 
- Fixed submit buttons to 'teal darken-4'
- GMaps are now "roadmap" style (aka not satellite images)
 - I did satellite images initally because I thought it would help them figure out where they put their camera. Similar to what @leian7 said
- On GMap, only one marker(google's red flag thing) will exist where the user clicks. The inital marker is on the site location
- Camera Point View has a GMap with a static marker on the camera point location
- Photo point now take's it's parent camera point on creation

athai commented:
So after creating and submitting the pp_form, the established date of the individual photo point image matches its camera point. However, if a different date is selected on the form, the photo is labelled with that date...

![datesdatesdates](https://cloud.githubusercontent.com/assets/6801364/25251921/ee771786-25cf-11e7-9b43-417b567b229e.png)

It's a good thing to keep, I think, since we do want to keep track of photos with differing dates. However, could you move the label down to be with the rest of the photo's info?

alxngyn commented:
@athai moved the image date down
![screenshot from 2017-04-21 13 59 21](https://cloud.githubusercontent.com/assets/7299081/25296027/1eacc646-269b-11e7-9204-1504af88de1a.png)

leian7 commented on streamwebs_frontend/streamwebs/templates/streamwebs/datasheets/camera_point_add.html:
```diff
@@ -5,28 +5,183 @@
+              {{ site.site_name }}
+          </a>
+        </h3>
+        <h4 align="center">{% trans 'New Photo Point Sampling' %}</h4>
```
Just "New Photo Point" is probably fine. It's not exactly a sampling.

leian7 commented:
+1

athai commented:
@leian7 Aside from adding a new photo, is there any other functionality we want upon editing a photo point? Currently, when you click on a specific photo point entry, you can add a new image. When navigating back to the main camera point page, the images appear stacked.
![stacked](https://cloud.githubusercontent.com/assets/6801364/25357463/b3158954-28f2-11e7-9c40-0cfe5325849d.png)

@alxngyn This looks good so far. Do you think the photo point thumbnail can be displayed differently if it contains multiple images? If it seems like it'd be a quickfix, can you add the change to your PR? Otherwise, please make a new issue so we don't forget about it.

alxngyn commented:
@leian7 fixed
@athai fixed. It will show the first image related to the photo point. I also removed the "image date" and replaced it with a "Number of images" field that shows the count.
![screenshot from 2017-04-26 14 12 10](https://cloud.githubusercontent.com/assets/7299081/25457530/8b5489c0-2a8b-11e7-9e40-6f3eee62eeeb.png)
